h2/src/main/org/h2/expression/function/ToChar.java

```diff
@@ -828,9 +828,32 @@ public static String toCharDateTime(Session session, Value value, String format,
             } else if (containsAt(format, i, "TZD") != null) {
                 output.append(getTimeZone(session, value, true));
                 i += 3;
+            } else if (containsAt(format, i, "TZH") != null) {
+                String result;
+                int hours = DateTimeFunctions.extractIntegerField(session, value, DateTimeFunctions.TIMEZONE_HOUR);
+                StringBuilder b = new StringBuilder();
```

Comment:
Please, remove `result` and `b`, there is no need for their allocation. You can append everything to the `output` directly.